File: programmers/김부건/2nd_week/target_number.js
function solution(numbers, target) {
	const dfs = (depth, result) => {
		if (depth === numbers.length) {
			if (result === target) {
				return 1;
			}
			return 0;
		}

		return dfs(depth + 1, result + numbers[depth]) + dfs(depth + 1, result - numbers[depth]);
	};

	return dfs(0, 0);
}

File: programmers/2022mock_exam/엄강우/2회/targetNumber.py
def make_num(sum_num, cnt, target, arr):
    result = 0
    if cnt == len(arr):
        if target == sum_num:
            return 1
        return 0 
    result += make_num(sum_num - arr[cnt], cnt + 1, target, arr)
    result += make_num(sum_num + arr[cnt], cnt + 1, target, arr)
    return result
    
def solution(numbers, target):
    answer = make_num(0, 0, target, numbers)
    return answer